File: clients/csharp/examples/tables-insert-one-blob.cs
using DataStax.AstraDB.DataApi;
using DataStax.AstraDB.DataApi.Tables;

namespace Examples;

public class ExampleRow
{
  [ColumnPrimaryKey]
  [ColumnName("title")]
  public string Title { get; set; } = null!;

  [ColumnName("example_blob")]
  public byte[]? ExampleBlob { get; set; }
}

public class Program
{
  static async Task Main()
  {
    // Get an existing table
    var client = new DataAPIClient();
    var database = client.GetDatabase(
      "**API_ENDPOINT**",
      "**APPLICATION_TOKEN**"
    );
    var table = database.GetTable<ExampleRow>("**TABLE_NAME**");

    // Insert a binary value
    var row = new ExampleRow()
    {
      ExampleBlob = new byte[]
      {
        0x3D,
        0xFB,
        0xE7,
        0x6D,
        0x3E,
        0xE9,
        0x78,
        0xD5,
        0x3F,
        0x49,
        0xFB,
        0xE7,
      },
      Title = "Example",
    };

    await table.InsertOneAsync(row);
  }
}

File: clients/java/examples/tables-insert-one-blob.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.tables.Table;
import com.datastax.astra.client.tables.commands.results.TableInsertOneResult;
import com.datastax.astra.client.tables.definition.rows.Row;

public class Example {

  public static void main(String[] args) {
    // Get and existing table
    Table<Row> table =
        new DataAPIClient("**APPLICATION_TOKEN**")
            .getDatabase("**API_ENDPOINT**")
            .getTable("**TABLE_NAME**");

    // Insert binary values
    byte[] exampleBytes = {
      (byte) 0x3D, (byte) 0xFB, (byte) 0xE7, (byte) 0x6D,
      (byte) 0x3E, (byte) 0xE9, (byte) 0x78, (byte) 0xD5,
      (byte) 0x3F, (byte) 0x49, (byte) 0xFB, (byte) 0xE7
    };

    Row row = new Row().addBlob("example_blob", exampleBytes).addText("title", "Example");

    TableInsertOneResult result = table.insertOne(row);
  }
}
